### PR TITLE
feat(ccm): add resource ccm private ca

### PR DESCRIPTION
**Pull request by saf3dfsa**
**What this PR does / why we need it**:
Add resource ccm private ca.

## PR Checklist

* [√] Tests added/passed.
* [√] Documentation updated.
* [√] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/ccm" TESTARGS="-run TestAccCCMPrivateCA_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ccm -v -run TestAccCCMPrivateCA_basic -timeout 360m -parallel 4
=== RUN   TestAccCCMPrivateCA_basic
=== PAUSE TestAccCCMPrivateCA_basic
=== CONT  TestAccCCMPrivateCA_basic
--- PASS: TestAccCCMPrivateCA_basic (18.49s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ccm       18.533s
```

### CODE REVIEW

File: huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"common_name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.All(
```
Comment (ShiChangkuo) [inline huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go]: `ValidateFunc` should be deleted, and add the limitation in docs.

File: huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourcePrivateCACreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if d.Get("type").(string) == "SUBORDINATE" {
+		if v := d.Get("issuer_id"); v == "" {
+			return diag.Errorf("error: required query parameter [%s] for subordinate CA is not set", "issuer_id")
```
Comment (ShiChangkuo) [inline huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go]: what does **query** mean?

File: huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourcePrivateCACreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if d.Get("type").(string) == "SUBORDINATE" {
+		if v := d.Get("issuer_id"); v == "" {
```
Comment (ShiChangkuo) [inline huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go]:
```suggestion
		if _, ok := d.GetOk("issuer_id"); !ok {
```

File: huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	var (
+		createPrivateCAHttpUrl = "v1/private-certificate-authorities"
+		createPrivateCAProduct = "ccm"
+	)
+	createPrivateCAClient, err := cfg.NewServiceClient(createPrivateCAProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating ccm Client: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go]:
```suggestion
		return diag.Errorf("error creating CCM client: %s", err)
```

File: huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go [hunk 5]
```diff
@@ -0,0 +1,514 @@
...
+	)
+	privateCAClient, err := cfg.NewServiceClient(privateCAProduct, region)
+	if err != nil {
+		return diag.Errorf("error deleting ccm Client: %s", err)
+	}
+	privateCAPath := privateCAClient.Endpoint + privateCAHttpUrl
+	privateCAPath = strings.ReplaceAll(privateCAPath, "{id}", d.Id())
+	privateCAOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
```
Comment (ShiChangkuo) [inline huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go:423]: actually, we can delete OkCodes

File: huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go [hunk 6]
```diff
@@ -0,0 +1,514 @@
...
+	status := utils.PathSearch("status", getPrivateCARespBody, nil).(string)
+	if !(status == "EXPIRED" || status == "DISABLE") {
+		disablePrivateCAPath := privateCAPath + "/disable"
+		_, err = privateCAClient.Request("POST", disablePrivateCAPath, &privateCAOpt)
+		if err != nil {
+			return diag.Errorf("error disabling private CA: %s", err)
+		}
+	}
+
+	pendingDays := "7"
```
Comment (ShiChangkuo) [inline huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go]: it's not suggest to set the default value of parameters. We should define it as **Required**

File: docs/resources/ccm_private_ca.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+* `crl_name` - (Optional, String, ForceNew) Specifies the name of the certificate revocation list.
+  It is [issuer_name] by default. Changing this parameter will create a new resource.
+
+* `obs_bucket_name` - (Required, String, ForceNew) Specifies the OBS bucket name.
+  Changing this parameter will create a new resource.
+
+* `valid_days` - (Required, Int, ForceNew) Specifies the CRL update interval, in days.It's limited between `7` to `30`.
+  Changing this parameter will create a new resource.
+
+## Attribute Reference
```
Comment (ShiChangkuo) [inline docs/resources/ccm_private_ca.md:163]: can the resource be imported?

File: huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_ca_test.go [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+func tesPrivateCA_basic(commonName string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_ccm_private_ca" "test_root" {
+  type = "ROOT"
+  distinguished_name {
+    common_name = "%s-root"
+    country = "CN"
+    state = "GD"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_ca_test.go]: format error: should align with "="

File: docs/resources/ccm_private_ca.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+  distinguished_name {
+    common_name         = "test-subordinate"
+    country             = "CN"
+    state               = "GD"
+    locality            = "SZ"
+    organization        = "huawei"
+    organizational_unit = "cloud"
+  }
+  key_algorithm       = "RSA2048"
+  issuer_id           = huaweicloud_ccm_private_ca.test_root.id
```
Comment (ShiChangkuo) [inline docs/resources/ccm_private_ca.md]: it's better to use a variable, the following is an example:
```hcl
variable "root_issuer_id" {}

resource "huaweicloud_ccm_private_ca" "test_subordinate" {
  issuer_id           = var.root_issuer_id
}
```

File: docs/resources/ccm_private_ca.md [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+## Argument Reference
+
+The following arguments are supported:
+
+* `type` - (Required, String, ForceNew) Specifies the type of private CA. Options are: **ROOT**, **SUBORDINATE**.
+  Changing this parameter will create a new resource.
+
+* `distinguished_name` - (Required, List, ForceNew) Specifies the distinguished name of private CA.
+  Changing this parameter will create a new resource.
+  The [distinguished_name] structure is documented below.
```
Comment (ShiChangkuo) [inline docs/resources/ccm_private_ca.md]:
```md
  The [distinguished_name](#block-distinguished_name) structure is documented below.
...
<a name="block-distinguished_name"></a>
```

File: docs/resources/ccm_private_ca.md [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+    type  = "DAY"
+    value = 5
+  }
+}
+```
+
+### create a suboridinate private CA
+
+```hcl
+resource "huaweicloud_ccm_private_ca" "test_subordinate" {
```
Comment (ShiChangkuo) [inline docs/resources/ccm_private_ca.md:40]: the required `pending_days` is missing.

File: huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_ca_test.go [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+					"validity", "key_usages", "pending_days",
+				},
+			},
+		},
+	})
+}
+
+func tesPrivateCA_basic(commonName string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_ccm_private_ca" "test_root" {
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_ca_test.go:108]: the region must be cn-north-4, please refer to https://github.com/huaweicloud/terraform-provider-huaweicloud/pull/3665

File: huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go [hunk 13]
```diff
@@ -0,0 +1,448 @@
...
+					},
+				},
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"id": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go]: `id` can be deleted.

**Comment by Lance52259** [comment]
pls update the supported table of the document `enterprise_project.md` if the new resource support `enterprise_project_id` parameter.

File: huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_ca_test.go [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+					resource.TestCheckResourceAttrSet(resourceName, "free_quota"),
+					resource.TestCheckResourceAttrSet(resourceName, "expired_at"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"validity", "key_usages", "pending_days", "region",
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_ca_test.go]: why do we ignore `region`?

**Pull request merged**